packages/e2e-tests/specs/writing-flow.test.js

```diff
@@ -141,6 +142,42 @@ describe( 'adding blocks', () => {
 		expect( await getEditedPostContent() ).toMatchSnapshot();
 	} );
 
+	it( 'should navigate around nested inline boundaries', async () => {
+		await setPostContent( `
+			<!-- wp:paragraph -->
+			<p><strong><em>1</em> <em>2</em></strong></p>
+			<!-- /wp:paragraph -->
+		` );
+
+		await page.click( '[data-type="core/paragraph"] [contenteditable]' );
+
+		expect( await getEditedPostContent() ).toMatchSnapshot();
+
+		await pressKeyTimes( 'ArrowLeft', 9 );
+
+		await page.keyboard.type( 'a' );
+		await page.keyboard.press( 'ArrowRight' );
+		await page.keyboard.type( 'b' );
+		await page.keyboard.press( 'ArrowRight' );
+		await page.keyboard.type( 'c' );
+		await page.keyboard.press( 'ArrowRight' );
+		await page.keyboard.type( 'd' );
+		await page.keyboard.press( 'ArrowRight' );
+		await page.keyboard.type( 'e' );
+		await page.keyboard.press( 'ArrowRight' );
+		await page.keyboard.type( 'f' );
+		await page.keyboard.press( 'ArrowRight' );
+		await page.keyboard.type( 'g' );
+		await page.keyboard.press( 'ArrowRight' );
+		await page.keyboard.type( 'h' );
+		await page.keyboard.press( 'ArrowRight' );
+		await page.keyboard.type( 'i' );
+		await page.keyboard.press( 'ArrowRight' );
+		await page.keyboard.type( 'j' );
+
+		expect( await getEditedPostContent() ).toMatchSnapshot();
+	} );
+
 	it( 'should clean TinyMCE content', async () => {
```

Comment:
This test case seems unnecessary now.

Also consider some comment updates to:

- https://github.com/WordPress/gutenberg/blob/b488d83b9f0863d01be4b6c558e23506a6f87a4d/packages/e2e-tests/specs/splitting-merging.test.js#L142
- https://github.com/WordPress/gutenberg/blob/b488d83b9f0863d01be4b6c558e23506a6f87a4d/packages/e2e-tests/specs/writing-flow.test.js#L94-L106

Reply:
In the WritingFlow component itself, I wonder about the relevance of these comments, and whether there's any need for implementation updates:

https://github.com/WordPress/gutenberg/blob/b488d83b9f0863d01be4b6c558e23506a6f87a4d/packages/editor/src/components/writing-flow/index.js#L240

https://github.com/WordPress/gutenberg/blob/b488d83b9f0863d01be4b6c558e23506a6f87a4d/packages/editor/src/components/writing-flow/index.js#L248-L255

https://github.com/WordPress/gutenberg/blob/b488d83b9f0863d01be4b6c558e23506a6f87a4d/packages/editor/src/components/writing-flow/index.js#L265-L269

Reply:
Yeah, I'll have a look through these.